maven-core/src/main/java/org/apache/maven/plugin/DefaultBuildPluginManager.java

```diff
@@ -134,7 +133,10 @@ public void executeMojo( MavenSession session, MojoExecution mojoExecution )
 
                 mojoExecutionListener.beforeMojoExecution( mojoExecutionEvent );
 
-                mojo.execute();
+                if ( mojoExecution.canRun( mojo, session ) )
```

Comment:
most of the modifications for this class seem to be purely formatting: restoring initial formatting would ease review